pyodide_build/tests/test_venv.py

```diff
@@ -0,0 +1,220 @@
+import os
+import shutil
+import subprocess
+
+import pytest
+
+from pyodide_build.out_of_tree import venv
+from pyodide_build.xbuildenv import CrossBuildEnvManager
+
+
+@pytest.fixture(scope="session")
+def base_test_dir(tmp_path_factory):
+    """Create a session-wide directory that persists for all tests."""
+    base_dir = tmp_path_factory.getbasetemp() / "pyodide_test_base"
+    base_dir.mkdir(exist_ok=True)
+
+    venv_path = base_dir / "test_venv"
+    venv_path.mkdir(exist_ok=True)
+
+    cwd = os.getcwd()
+    os.chdir(str(base_dir))
+
+    xbuildenv_test_name = ".pyodide-xbuildenv-for-testing"
+
+    manager = CrossBuildEnvManager(xbuildenv_test_name)
+    manager.install()
```

Comment:
> The CI should pass now, but if you'd like me to label the slow tests as integration tests, I can do that.

Yeah, maybe let's mark tests that rely on this fixture as an integration tests, as I think it is a good to not rely on real xbuildenv in the unittest.

Also calling `install` without the argument assumes that the version will be properly set in the local environment, which is also a bit fragile IMO. So how about passing the URL to the absolute xbuildenv URL instead?

Reply:
Fair point – done in commits bf6c872538b1dd0fa96067f2adb924c7bacb9d37 and 6ee83e9c6485f61ede86b051430f5fea346caf66. As we now have two kind of integration tests, where we run the recipes and now some tests that are code-based, I had to split the coverage collection. In another PR, I think we could write a minimal test driver file that would make `pytest` run the `make test-recipe` command, or switch to a task runner designed to work with `pytest`, like [`nox`](https://nox.thea.codes/en/stable/).